Failing container image build in GitHub Actions

```
> [build 5/8] RUN dotnet restore PoshMcp.sln     && dotnet nuget locals http-cache --clear:
0.541 /usr/share/dotnet/sdk/10.0.202/NuGet.targets(519,5): error MSB3202: The project file "/src/TestClient/TestClient.csproj" was not found. [/src/PoshMcp.sln]
0.541 /usr/share/dotnet/sdk/10.0.202/NuGet.targets(519,5): error MSB3202: The project file "/src/PoshMcp.Tests/PoshMcp.Tests.csproj" was not found. [/src/PoshMcp.sln]
------
Dockerfile:14
--------------------
  13 |     # This layer is cached unless project files change
  14 | >>> RUN dotnet restore PoshMcp.sln \
  15 | >>>     && dotnet nuget locals http-cache --clear
  16 |     
--------------------
ERROR: failed to build: failed to solve: process "/bin/sh -c dotnet restore PoshMcp.sln     && dotnet nuget locals http-cache --clear" did not complete successfully: exit code: 1
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -11,7 +11,7 @@ COPY PoshMcp.Server/PoshMcp.csproj ./PoshMcp.Server/
 
 # Restore dependencies in one stage for efficiency
 # This layer is cached unless project files change
-RUN dotnet restore PoshMcp.sln \
+RUN dotnet restore PoshMcp.Server/PoshMcp.csproj \
     && dotnet nuget locals http-cache --clear
 
 # Copy all source code (changes frequently - placed after stable dependencies)
@@ -19,7 +19,7 @@ COPY . .
 
 # Build and publish the PoshMcp.Server application in Release configuration
 # Use /p:UseAppHost=false for portability across architectures
-RUN dotnet build PoshMcp.sln -c Release --no-restore \
+RUN dotnet build PoshMcp.Server/PoshMcp.csproj -c Release --no-restore \
     && dotnet publish PoshMcp.Server/PoshMcp.csproj -c Release -o /app/publish/server /p:UseAppHost=false
 
 # Copy startup script to publish output

--- a/Dockerfile
+++ b/Dockerfile
@@ -5,8 +5,7 @@ FROM mcr.microsoft.com/dotnet/sdk:10.0 AS build
 
 WORKDIR /src
 
-# Copy solution and project files first (stable layer - benefits from layer caching)
-COPY PoshMcp.sln ./
+# Copy project files first (stable layer - benefits from layer caching)
 COPY PoshMcp.Server/PoshMcp.csproj ./PoshMcp.Server/
 
 # Restore dependencies in one stage for efficiency

